src/estimation/src/BerdyHelper.cpp

```diff
@@ -1366,6 +1401,35 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
         // bY for the joint wrenches is zero
     }
 
+    ////////////////////////////////////////////////////////////////////////
+    ///// COM ACCELERATION SENSOR
+    ////////////////////////////////////////////////////////////////////////
+    if (m_options.includeCoMAccelerometerAsSensor)
+    {
+        // Get the row index corresponding to the com accelerometer sensor
+        IndexRange comAccelerometerRange = this->getRangeCoMAccelerometerSensorVariable(COM_ACCELEROMETER_SENSOR);
+
+        for(LinkIndex idx = 0; idx < static_cast<LinkIndex>(m_options.comConstraintLinkIndexVector.size()); idx++)
```

Comment:
When you are considering all the link this is probably not a problem because the index of the comConstraintLinkIndexVector most likely correspond to the index of the link. But if it is not the case, this can cause problems.

Reply:
@lrapetti Great catch!
I fixed it in https://github.com/robotology/idyntree/pull/571/commits/8b9ffe5c2fc2d8b718da6a8a4fec0a2780eb6c5e
Please check it and resolve it